ArcGISRuntimeSDKQt_CppSamples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.cpp

```diff
@@ -0,0 +1,244 @@
+// [WriteFile Name=AddClusteringFeatureReductionToAPointFeatureLayer, Category=Layers]
+// [Legal]
+// Copyright 2023 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp" // IWYU pragma: keep
+#endif // PCH_BUILD
+
+#include "AddClusteringFeatureReductionToAPointFeatureLayer.h"
+
+#include "ClassBreaksRenderer.h"
+#include "SimpleMarkerSymbol.h"
+#include "SymbolTypes.h"
+#include "ClusteringFeatureReduction.h"
+#include "AggregateField.h"
+#include "AggregateFieldListModel.h"
+#include "ReductionTypes.h"
+#include "FeatureLayer.h"
+#include "PopupDefinition.h"
+#include "Error.h"
+#include "LayerListModel.h"
+#include "SimpleLabelExpression.h"
+#include "TextSymbol.h"
+#include "LabelDefinitionListModel.h"
+#include "LabelDefinition.h"
+#include "ServiceTypes.h"
+#include "IdentifyLayerResult.h"
+#include "Popup.h"
+#include "PopupManager.h"
+
+#include <QColor>
+
+using namespace Esri::ArcGISRuntime;
+
+AddClusteringFeatureReductionToAPointFeatureLayer::AddClusteringFeatureReductionToAPointFeatureLayer(QObject* parent /* = nullptr */):
+  QObject(parent)
+{
+  // Create a map from a web map PortalItem.
+  m_map = new Map(new PortalItem("aa44e79a4836413c89908e1afdace2ea", this), this);
+
+  // Get the Zurich buildings feature layer once the map has finished loading.
+  connect(m_map, &Map::doneLoading, this, [this](const Error& error)
+          {
+            if (!error.isEmpty())
+            {
+              qDebug() << "Map loading error:" << error.message() << error.additionalMessage();
+              return;
+            }
+
+            m_layer = static_cast<FeatureLayer*>(m_map->operationalLayers()->first());
```

Comment:
I recommend writing defensive code here in to check that both there is at least 1 layer, and the first layer is a FeatureLayer. Should anything go change with the service in the future, nothing bad will happen with the sample.